From e1de25fefbb324e6ceaae29a0454a04c96b72158 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 2 Sep 2014 15:09:46 -0400
Subject: [redhat] minor cleanup of rh_check_supported() for now-supported CPUs

Message-id: <1409670586-4308-1-git-send-email-prarit@redhat.com>
Patchwork-id: 90653
O-Subject: [RHEL7.1 PATCH BZ 1136448] x86, minor cleanup of rh_check_supported()
Bugzilla: 1136448
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

Processor models 61, 62, and 63 are supported.  Cleanup the code.

Successfully tested by me.

RHEL_only.

Cc: Vivek Goyal <vgoyal@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Don Zickus <dzickus@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/setup.c b/arch/x86/kernel/setup.c
index 79ec977..c9b7040 100644
--- a/arch/x86/kernel/setup.c
+++ b/arch/x86/kernel/setup.c
@@ -870,12 +870,9 @@ static void rh_check_supported(void)
   switch (boot_cpu_data.x86_model) {
   case 77: /* Atom Avoton */
   case 70: /* Crystal Well */
-  case 63: /* Grantley/Haswell EP */
-  case 62: /* Ivy Town */
-  case 61: /* Broadwell */
    break;
   default:
-   if (boot_cpu_data.x86_model > 60) {
+   if (boot_cpu_data.x86_model > 63) {
     printk(KERN_CRIT
            "Detected CPU family %d model %d\n",
            boot_cpu_data.x86,
-- 
1.7.1